p2p/crypto/secio/rw.go

```diff
@@ -130,6 +134,10 @@ func (r *etmReader) Read(buf []byte) (int, error) {
 		return 0, err
 	}
 
+	if fullLen > MaxMsgSize {
+		return 0, ErrMaxMessageSize
+	}
```

Comment:
oh this makes sense. we may want to bump it even lower. what's going on here is that the msgio may receive random input. 

what's weird to me is that this happens _after_ encryption has been setup correctly. it would make more sense if we were seeing this before the handshake finished-- meaning we're speaking to something else entirely. 

This problem suggests we should use a well-defined small header as the first message (i.e. like the protocol-stream idea), before reading "length prefixes" from the network. (that, or cap the lengths to a reasonable size always. (e.g. handshake messages should be below 4KB)